### PR TITLE
refactor implicitConvToWithoutAliasThis()

### PR DESCRIPTION
**Pull request by WalterBright**
There was a lot of intricate logic that wasn't necessary.

### CODE REVIEW

File: compiler/src/dmd/dcast.d [hunk 1]
```diff
@@ -69,7 +69,7 @@ Expression implicitCastTo(Expression e, Scope* sc, Type t)
 {
     Expression visit(Expression e)
     {
-        // printf("Expression.implicitCastTo(%s of type %s) => %s\n", e.toChars(), e.type.toChars(), t.toChars());
```
Comment (thewilsonator) [inline compiler/src/dmd/dcast.d:72]: look unrelated

**Review by thewilsonator — APPROVED** [review]
Otherwise looks good

File: compiler/src/dmd/dcast.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/dcast.d:72]: it is unrelated, but it's trivial
